Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -2146,6 +2241,57 @@ var stagger = function(
   }));
 };
 
+type LoopAnimationConfig = { iterations: number };
+
+var loop = function(
+  animation: CompositeAnimation,
+  config: LoopAnimationConfig = {iterations: -1},
+): CompositeAnimation {
+  var isFinished = false;
+  var iterationsSoFar = 0;
+  // Config without iterations specified (e.g. only 'useNativeDriver') should loop indefinitely
+  if (!config.hasOwnProperty('iterations')) config.iterations = -1;
```

Comment:
You can avoid this check completely if you use destructuring + default parameters for the config object.

It should look like
```js
var loop = function(
  animation: CompositeAnimation,
  { iterations = -1 }: LoopAnimationConfig = {},
 )
```
This way it works whether you call it with `loop(anim)` or `loop(anim, {})`.